micro-rdk/src/common/conn/viam.rs

```diff
@@ -454,6 +464,9 @@ where
     }
 
     pub(crate) async fn run(&mut self) -> ! {
+        if let Err(err) = self.storage.log_space_diagnostic() {
```

Comment:
I'm not sure it makes sense for `log_space_diagnostic` to return an error at all. If it can't log, it is unlikely that this can either, and if it can log, it can report its own failures.